test/e2e/retry.go

```diff
@@ -0,0 +1,53 @@
+// +build e2e
+
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"k8s.io/apimachinery/pkg/util/wait"
+	"time"
+)
+
+const (
+	// Parameters for retrying with exponential backoff.
+	retryBackoffInitialDuration = 100 * time.Millisecond
+	retryBackoffFactor          = 3
+	retryBackoffJitter          = 0.1
+	retryBackoffSteps           = 3
+)
+
+// retryWithExponentialBackOff retries the function until it returns nil,
+// or until the number of attempts (steps) has reached the maximum value.
+func retryWithExponentialBackOff(fn func() error) error {
```

Comment:
nice! @shysank

Reply:
Thanks! Although I was heavily inspired (shamelessly copied) by [this](https://github.com/kubernetes-sigs/cluster-api/blob/e6b0039b4aa1021b0a9afec8a527d969c20acabb/cmd/clusterctl/client/cluster/client.go#L238) :)

Reply:
lol, that's okay, you still used it nicely